crates/pink/runtime/src/storage/mod.rs

```diff
@@ -90,3 +95,28 @@ where
 }
 
 pub mod external_backend;
+
+pub fn maybe_emit_system_event_block(events: SystemEvents) {
+    use pink_capi::v1::ocall::OCalls;
+    if events.is_empty() {
+        return;
+    }
+    if !OCallImpl.exec_context().mode.is_transaction() {
+        return;
+    }
+    let body = EventsBlockBody {
+        phala_block_number: System::block_number(),
```

Comment:
Why do you put it in the body, the header?

Reply:
It was in header. I moved it to body because as a `block chain header`, the current fields contains all necessary information that required to validate the chain. Some services can only store/decode the headers, other info should be the payload of the chain, and can be decoded on demand.

Reply:
Make sense